Sming/Components/mdns/src/Packet.h

```diff
@@ -0,0 +1,127 @@
+/****
+ * Sming Framework Project - Open Source framework for high efficiency native ESP8266 development.
+ * Created 2015 by Skurydin Alexey
+ * http://github.com/SmingHub/Sming
+ * All files of the Sming Core are provided under the LGPL v3 license.
+ *
+ * Packet.h
+ *
+ ****/
+
+#pragma once
+
+#include <WString.h>
+
+namespace mDNS
+{
+/**
+ * @brief Helper class for reading/writing packet content
+ */
+struct Packet {
+	uint8_t* data;
+	mutable uint16_t pos;
+
+	const uint8_t* ptr() const
+	{
+		return data + pos;
+	}
+
+	uint8_t* ptr()
+	{
+		return data + pos;
+	}
+
+	void skip(uint16_t len) const
+	{
+		pos += len;
+	}
+
+	uint8_t peek8() const
+	{
+		return data[pos];
+	}
+
+	uint8_t read8() const
+	{
+		return data[pos++];
+	}
+
+	uint16_t peek16() const
+	{
+		return (data[pos] << 8) | data[pos + 1];
+	}
+
+	uint16_t read16() const
+	{
+		return (read8() << 8) | read8();
+	}
+
+	uint32_t read32() const
+	{
+		return (read16() << 16) | read16();
+	}
+
+	void read(void* buffer, uint16_t len) const
+	{
+		memcpy(buffer, ptr(), len);
+		pos += len;
+	}
+
+	String readString(uint16_t length) const
+	{
+		String s;
+		if(s.setLength(length)) {
+			read(s.begin(), length);
+		} else {
+			pos += length;
+		}
+		return s;
+	}
+
+	void write8(uint8_t value)
+	{
+		data[pos++] = value;
+	}
+
+	void write16(uint16_t value)
+	{
+		write8(value >> 8);
+		write8(value & 0xff);
+	}
+
+	void write32(uint32_t value)
+	{
+		write16(value >> 16);
+		write16(value & 0xffff);
+	}
+
+	void write(const void* s, uint16_t len)
+	{
+		memcpy(ptr(), s, len);
+		pos += len;
+	}
+
+	void writeName(const String& name)
```

Comment:
The Packet stuff without the specific `writeName(...)` method could be moved to a more general location like Core/Data/ so that it can be used also from other places.

Reply:
Good idea. `Packet` is a pretty crap name though, any suggestions?

Reply:
> any suggestions?

Go with `Packet` for now. 

At some later PR(s) we should try to unify a bit the DNS and mDNS parsing

Reply:
It should probably go into `/Core/Network/WebHelpers`

Reply:
Possible but if we start using it in other non-network related code then it should move to `Core/Data/`.

Reply:
It's specifically because it deals with network byte order.

Reply:
If we use it for more general use then we might expect it to be able to write in both LSB-first (host order) and MSB-first (Network order) so renaming methods such as `write16` -> `write16n`, or adding `write16h`, etc.

Reply:
Happy to that, I'm sure it would be useful.

Reply:
Or, on reflection, two separate versions called `NetworkPacket` and `HostPacket` would be better I think.